teps/0091-trusted-resources.md

```diff
@@ -364,26 +372,127 @@ type KeylessRef struct {
-1) Return error in reconciler, mark the status as error with `FailedVerification` reason;
-2) Not return error, but mark the TaskRun/PipelineRun as `FailedVerification`;
-3) Skip the verification;
-
-Configuration sample:
-```yaml
+ ```yaml
 apiVersion: v1
 kind: ConfigMap
 metadata:
   name: feature-flag
   namespace: tekton-pipelines
 data:
-  verification-policy: enforce
+  verification-no-match-policy: "deny"
 ```
 
-`verification-policy`. (Optional, `enforce`, `warn` or `skip`, default `skip`):
- * `enforce`: Failing verification will mark the taskruns/pipelineruns as failed.
- * `warn`: Log warning but don't fail the taskruns/pipelineruns.
- * `skip`: Directly skip the verification.
+`verification-no-match-policy`. (Optional, `allow`, `warn` or `deny`, default to `allow`):
+ * `allow`: Don't fail the taskrun/pipelinerun if no matching policies are found. Don't log.
+ * `warn`: Don't fail the taskrun/pipelinerun and log a warning if no matching policies are found.
+ * `deny`: Fail the taskrun/pipelinerun if no matching policies are found.
+
+ Reconciler will respond to verification failure based on ConfigMap and Verification Policy exsitence settings to:
+1) Return error in reconciler, mark the status as error with `FailedVerification` reason if `enforce` policies fail
+2) Not return error, but warn the log and update the status condition with `FailedVerification` reason if only found `warn` policies.
+3) Skip the verification if no policies exist and `verification-no-match-policy` is set to `allow`.
+
+**Note:** The current proposed `verification-no-match-policy` will be added to replace the old `resource-verification-mode` in one release and this is not a backwards compatible change.
+
+ **How configmap works with Verification Policy**
+ * If the `verification-no-match-policy` is set to `deny`
+   * No Matching Policies: Resources that do not have matched policies will fail the taskrun/pipelinerun.
+   * Matching Policies:
+     * Don't fail taskrun/pipelinerun if all the matched `enforce` policies pass or only match `warn` policies.
+     * Fail the taskrun/pipelinerun if any matched `enforce` policy fails to verify.
+
+ * If the `verification-no-match-policy` is set to `allow` or `warn`,
+   * No Matching Policies: Don't fail taskrun/pipelinerun, and will emit warning log if set to `warn`.
+   * Matching Policies: Same as `deny`.
+
+
+ #### Condition Update
+
+ Trusted resources should also update the taskrun/pipelinerun’s condition to indicate if it passes verification or not. This can be done via knative api. Other options are listed in [alternatives](#alternatives).
+
+ Need to create conditions for trusted resources so this can co-exist with current `apis.ConditionSucceeded` type:
+ ```go
+ const (
+	// ConditionVerification specifies that the resources pass trusted resources verification or not.
+	ConditionVerification apis.ConditionType = "TrustedResourcesVerification"
```

Comment:
sure! sgtm!

Reply:
found [conditions conventions](https://github.com/kubernetes/community/blob/4c9ef2d/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties) that back this request to rename the condition:

> Condition type names should describe the current observed state of the
  resource, rather than describing the current state transitions. This
  typically means that the name should be an adjective ("Ready", "OutOfDisk")
  or a past-tense verb ("Succeeded", "Failed") rather than a present-tense verb
  ("Deploying"). Intermediate states may be indicated by setting the status of
  the condition to `Unknown`.